src/prefect/blocks/system.py

```diff
@@ -86,25 +101,27 @@ class DateTime(Block):
     _logo_url = "https://cdn.sanity.io/images/3ugk85nk/production/8b3da9a6621e92108b8e6a75b82e15374e170ff7-48x48.png"
     _documentation_url = "https://docs.prefect.io/api-ref/prefect/blocks/system/#prefect.blocks.system.DateTime"
 
-    value: DateTime = Field(
+    value: PydanticDateTime = Field(
         default=...,
         description="An ISO 8601-compatible datetime value.",
     )
 
 
-class Secret(Block):
+class Secret(Block, Generic[T]):
     """
     A block that represents a secret value. The value stored in this block will be obfuscated when
-    this block is logged or shown in the UI.
+    this block is viewed or edited in the UI.
 
     Attributes:
-        value: A string value that should be kept secret.
+        value: A value that should be kept secret.
 
     Example:
         ```python
         from prefect.blocks.system import Secret
 
-        secret_block = Secret.load("BLOCK_NAME")
+        Secret(value="sk-1234567890").save("test-secret", overwrite=True)
```

Comment:
```suggestion
        Secret(value="sk-1234567890").save("BLOCK_NAME", overwrite=True)
```

Reply:
`BLOCK_NAME` is a placeholder in the UI and gets replaced by the user's block name, so we might want to leave it in for now.